pkg/operator/controller/nodeobservability/controller.go

```diff
@@ -287,3 +293,17 @@ func (r *NodeObservabilityReconciler) ensureNodeObservabilityDeleted(ctx context
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+func reconcileClusterNamedResource() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return strings.EqualFold(nodeObsCRName, e.Object.GetName())
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return strings.EqualFold(nodeObsCRName, e.ObjectNew.GetName())
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return strings.EqualFold(nodeObsCRName, e.Object.GetName())
+		},
+	}
```

Comment:
[NewPredicateFuncs](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/predicate#NewPredicateFuncs) can reduce some boilerplace, plus it implements the missing `GenericFunc`.